File: SDE_sheet/Arrays/06_stock_buy_and_sell/solution.go
func maxProfit(prices []int) int {
    if len(prices) == 0 {
        return 0
    }
    
    maxProfit := 0

    for i := 1; i < len(prices); i++ {
        if prices[i] < prices[0] {
            prices[0] = prices[i]
        } else {
            profit := prices[i] - prices[0]
            if profit > maxProfit {
                maxProfit = profit
            }
        }
    }

    return maxProfit
}

File: SDE_sheet/Arrays/06_stock_buy_and_sell/solution.rs
impl Solution {
    pub fn max_profit(mut prices: Vec<i32>) -> i32 {
        if prices.is_empty() {
            return 0;
        }
        
        let mut max_profit = 0;

        for i in 1..prices.len() {
            if prices[i] < prices[0] {
                prices[0] = prices[i];
            } else {
                let profit = prices[i] - prices[0];
                if profit > max_profit {
                    max_profit = profit;
                }
            }
        }

        max_profit
    }
}

